src/Npgsql/TypeMapping/GlobalTypeMapper.cs

```diff
@@ -88,14 +76,41 @@ public override INpgsqlTypeMapper AddMapping(NpgsqlTypeMapping mapping)
             }
         }
 
+        void UpdateNonMappingTables(NpgsqlTypeMapping mapping)
+        {
+            if (mapping.NpgsqlDbType.HasValue)
+            {
+                _npgsqlDbTypeToPgTypeName[mapping.NpgsqlDbType.Value] = mapping.PgTypeName;
+                _npgsqlDbTypeToPgTypeName[mapping.NpgsqlDbType.Value | NpgsqlDbType.Array] = mapping.PgTypeName + "[]";
+
+                foreach (var dbType in mapping.DbTypes)
+                    _dbTypeToNpgsqlDbType[dbType] = mapping.NpgsqlDbType.Value;
+
+                if (mapping.InferredDbType.HasValue)
+                    _npgsqlDbTypeToDbType[mapping.NpgsqlDbType.Value] = mapping.InferredDbType.Value;
+
+                foreach (var clrType in mapping.ClrTypes)
+                {
+                    _typeToNpgsqlDbType[clrType] = mapping.NpgsqlDbType.Value;
+                    _typeToPgTypeName[clrType] = mapping.PgTypeName;
+                }
+            }
+
+            if (mapping.InferredDbType.HasValue)
+                foreach (var clrType in mapping.ClrTypes)
+                    _typeToDbType[clrType] = mapping.InferredDbType.Value;
+        }
+
         public override bool RemoveMapping(string pgTypeName)
         {
             Lock.EnterWriteLock();
             try
             {
-                var result = base.RemoveMapping(pgTypeName);
+                var oldMappingsByName = MappingsByName;
+                MappingsByName = MappingsByName.Remove(pgTypeName);
+                var changed = ReferenceEquals(MappingsByName, oldMappingsByName);
                 RecordChange();
```

Comment:
Maybe we shouldn't call `RecordChange` if we didn't actually remove anything?

Reply:
Makes sense.

Reply:
BTW RemoveMappings also does not remove from MappingsByNpgsqlDbType and MappingsByClrType :(